src/edu/washington/escience/myria/api/PerfEnforceResource.java

```diff
@@ -0,0 +1,131 @@
+package edu.washington.escience.myria.api;
+
+import java.util.List;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.glassfish.jersey.media.multipart.FormDataParam;
+import org.slf4j.LoggerFactory;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.api.encoding.PerfEnforceQueryMetadataEncoding;
+import edu.washington.escience.myria.api.encoding.PerfEnforceTableEncoding;
+import edu.washington.escience.myria.parallel.Server;
+import edu.washington.escience.myria.perfenforce.PerfEnforceException;
+
+/**
+ * This is the class that handles API calls for PerfEnforce
+ */
+@Path("/perfenforce")
+public final class PerfEnforceResource {
+
+  @Context private Server server;
+
+  /** Logger. */
+  protected static final org.slf4j.Logger LOGGER =
+      LoggerFactory.getLogger(PerfEnforceResource.class);
+
+  /**
+   * Begins generating a PSLA based on the input schema
+   *
+   * @param tableList describes all the tables and schema of the dataset
+   */
+  @POST
+  @Path("/preparePSLA")
+  public Response prepareData(final List<PerfEnforceTableEncoding> tableList)
+      throws PerfEnforceException, DbException, Exception {
+    server.getPerfEnforceDriver().preparePSLA(tableList);
+    return Response.noContent().build();
```

Comment:
I think this maps to HTTP return code 204. I'm not sure if 201 is more appropriate, if this POST creates any resource then 201 sounds better.